scripts/west_commands/runners/core.py

```diff
@@ -144,23 +145,30 @@ def _init(self):
         self._parse(self.path)
 
     def _parse(self, filename: str):
-        with open(filename, 'r') as f:
-            for line in f:
-                line = line.strip()
-                if not line or line.startswith('#'):
-                    continue
-                option, value = line.split('=', 1)
-                self.options[option] = self._parse_value(value)
-
-    @staticmethod
-    def _parse_value(value):
-        if value.startswith('"') or value.startswith("'"):
-            return value.split()
-        try:
-            return int(value, 0)
-        except ValueError:
-            return value
+        opt_value = re.compile('^(?P<option>CONFIG_[A-Za-z0-9_]+)=(?P<value>.*)$')
```

Comment:
Just a minor observation on changed behavior from existing code.

Previously, `line.strip()` was called, and all whitespaces was stripped.
This meant that a user updating `.config` by hand and by mistake introduce trailing withespaces like:
`CONFIG_TEST=y  `  would not see any problems, but with this PR, the value will now be:
`'y  '` instead of `'y'`.

I won't block the PR over this, but knowing how easy it is to accidentally introduce a stray `' '` when editing a file, and how hard it can be to trace back why the `.config` suddenly fails, I wonder if we should keep the strip.

Also the `value.endswith('"'):` further down won't match if there are trailing whitespaces.

Reply:
`.rstrip()` added.